pkg/test/framework/examples/example.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package examples
```

Comment:
AFAICT this isn't part of the test framework. Should it just be under `pkg/test/examples`?

Reply:
It was my suggestion to put it here. My  reasoning is that I think this might need some integration with the main test framework because ultimately in order to run these examples, it will be necessary to setup test environments and all that stuff, just like we do for normal tests.

WDYT?

Reply:
Ah ok, thanks for the background.  So which way do you see the dependency? test framework depends on examples? or examples on test framework?

If the latter, I see the structure similar to the way the conformance is a layer on top of the test framework, which is under `pkg/test/conformance`.

Reply:
I think examples should depend on test framework. I'll move this to pkg/test.

Reply:
Does it make sense to follow the model here: https://github.com/istio/istio/tree/master/tests/integration. There's a base package (lets say docs, examples is a bit vague at this level), which contains a framework directory (we can call this framework or examples). And other directories which contain actual tests.

Reply:
@brian-avery yeah I think some directory under `pkg/tests` makes sense.  Maybe we just be explicit and call it `pkg/tests/istio.io/examples`?

Reply:
sounds good to me.